src/autoscaler/metricsforwarder/forwarder/syslog_emitter.go

```diff
@@ -0,0 +1,102 @@
+package forwarder
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/metricsforwarder/config"
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/models"
+	"code.cloudfoundry.org/go-loggregator/v9/rpc/loggregator_v2"
+	"code.cloudfoundry.org/lager/v3"
+	"code.cloudfoundry.org/loggregator-agent-release/src/pkg/egress"
+	"code.cloudfoundry.org/loggregator-agent-release/src/pkg/egress/syslog"
+)
+
+type SyslogEmitter struct {
+	logger lager.Logger
+	Writer egress.WriteCloser
+}
+
+type Counter struct{}
+
+func (c *Counter) Add(delta float64) {
+}
+func (c *Counter) Set(delta float64) {
+}
+
+func NewSyslogEmitter(logger lager.Logger, conf *config.Config) (MetricForwarder, error) {
+	var writer egress.WriteCloser
+	var protocol string
+
+	tlsConfig, _ := conf.SyslogConfig.TLS.CreateClientConfig()
+
+	netConf := syslog.NetworkTimeoutConfig{
+		WriteTimeout: time.Second,
+		DialTimeout:  100 * time.Millisecond,
+	}
+
+	if conf.SyslogConfig.TLS.CACertFile != "" {
+		protocol = "syslog-tls"
+	} else {
+		protocol = "syslog"
+	}
+
+	url, _ := url.Parse(fmt.Sprintf("%s://%s:%d", protocol, conf.SyslogConfig.ServerAddress, conf.SyslogConfig.Port))
+
+	logger.Info("syslog-emitter-config", lager.Data{"url": url})
+
+	hostname, _ := os.Hostname()
+
+	binding := &syslog.URLBinding{
+		URL:      url,
+		Hostname: hostname,
+	}
+
+	switch binding.URL.Scheme {
+	case "syslog":
+		writer = syslog.NewTCPWriter(
+			binding,
+			netConf,
+			&Counter{},
+			syslog.NewConverter(),
+		)
+	case "syslog-tls":
+		writer = syslog.NewTLSWriter(
+			binding,
+			netConf,
+			tlsConfig,
+			&Counter{},
+			syslog.NewConverter(),
+		)
+	}
+
+	return &SyslogEmitter{
+		Writer: writer,
+		logger: logger,
+	}, nil
+}
+
+func (mf *SyslogEmitter) EmitMetric(metric *models.CustomMetric) {
+	e := &loggregator_v2.Envelope{
+		InstanceId: fmt.Sprintf("%d", metric.InstanceIndex),
+		Timestamp:  time.Now().UnixNano(),
+		SourceId:   metric.AppGUID,
+		Message: &loggregator_v2.Envelope_Gauge{
+			Gauge: &loggregator_v2.Gauge{
+				Metrics: map[string]*loggregator_v2.GaugeValue{
+					metric.Name: {
+						Unit:  metric.Unit,
+						Value: metric.Value,
+					},
+				},
+			},
+		},
+	}
+
+	err := mf.Writer.Write(e)
+	if err != nil {
+		mf.logger.Error("failed-to-write-metric-to-syslog", err)
+	}
```

Comment:
logging the errors in case of write failures is untested. shall we test this to ensure that we always have this logging place? we don't have error-logs in case this somehow gets deleted which makes bug-hunts super difficult : D 

I would prefer testing this as well.

WDYT?

Reply:
Another thing that came into my mind is the observability of connection errors to syslog. 

@silvestre do you know if Dynatrace is able to monitor the syslog protocol out of the box so that the SRE colleagues can easily write alerts that fire whenever we face problems emitting envelopes to syslog-server?

Reply:
@geigerj0 : I don't think that basic TCP connections would be picked up Dynatrace, as it is not explicitly [mentioned](https://docs.dynatrace.com/docs/setup-and-configuration/technology-support#go).

Let's keep this in the back of our mind and ensure that the errors are written if there are issues - in the end this is not a regression as previously we wouldn't have seen connection between `loggr-syslog-agent` and `log-cache` either.

Reply:
Ouch :/ not sure if this is a good thing because it literally means that we don't get any notification if the connection is somehow failing. 

Shall we create a backlog item to get prometheus metrics like `metricsforwarder_send_syslog_errors_count` somehow into Dynatrace, so that we can create a notification if the counter grows too fast? 🤓